hive_agent/server/routes/files.py

```diff
@@ -55,6 +56,7 @@ async def create_upload_files(files: List[UploadFile] = File(...)):
                     logger.info("Inserting data to existing basic index")
                     logger.info(f"Index: {index_store.list_indexes()}")
                     index_store.save_to_file()
+                    hive_agent.recreate_agent()
```

Comment:
is this function only used in tests

Reply:
it is used both on hive-agent-py/hive_agent/server/routes/files.py and  hive-agent-py/tests/test_agent.py.

Reply:
We need to reinit the agent with new tools and tool_retriever object